Modules/Core/Common/include/itkExceptionObject.h

```diff
@@ -57,10 +58,10 @@ class ITKCommon_EXPORT ExceptionObject : public std::exception
                            unsigned int lineNumber = 0,
                            const char * desc = "None",
                            const char * loc = "Unknown");
-  explicit ExceptionObject(const std::string & file,
-                           unsigned int        lineNumber = 0,
-                           const std::string & desc = "None",
-                           const std::string & loc = "Unknown");
+  explicit ExceptionObject(std::string  file,
```

Comment:
Note,  This change is compatible with the existing public API, and it allows for move semantics.   This change facilitates using itk::ExceptionObject as a base class.